server/model/monitor.js

```diff
@@ -395,19 +397,28 @@ class Monitor extends BeanModel {
                             }
                             const lastBeat = await Monitor.getPreviousHeartbeat(child.id);
 
-                            // Only change state if the monitor is in worse conditions then the ones before
+                            // Only change state if the monitor is in worse conditions than the ones before
                             // lastBeat.status could be null
                             if (!lastBeat) {
                                 bean.status = PENDING;
                             } else if (bean.status === UP && (lastBeat.status === PENDING || lastBeat.status === DOWN)) {
                                 bean.status = lastBeat.status;
+                                failChild = child.name;
                             } else if (bean.status === PENDING && lastBeat.status === DOWN) {
                                 bean.status = lastBeat.status;
+                                failChild = child.name;
+                            }
+                            if (lastBeat.status === DOWN) {
+                                failCount++;
```

Comment:
I think this if statement is ignoring the retry logic above => please integrate it above

Reply:
If there is a logic error, please describe a test case to demonstrate how the count could be incorrect. I can't see how that would happen. The retries are handled by the PENDING state, if I understand the logic correctly. By only checking DOWN here, I think the logic is fine.

Reply:
While it might be correct after longer looking at it, it is not at first looking at it.
=> not great for code-readability
=> should be integrated in the above loop as proposed in https://github.com/louislam/uptime-kuma/pull/4371#discussion_r1452822429